pandas/_libs/tslibs/offsets.pyx

```diff
@@ -3597,6 +3641,12 @@ cdef class FY5253Mixin(SingleConstructorOffset):
         self.variation = state.pop("variation")
 
     def is_anchored(self) -> bool:
+        warnings.warn(
+            f"{type(self).__name__}.is_anchored is deprecated and will be removed "
+            f"in a future version, please use \'obj.n == 1\' instead.",
```

Comment:
do we not want to say about `self.startingMonth is not None and self.weekday is not None` in this one?

Reply:
I am not sure about this. I think, there is no need to check if parameters `startingMonth` and `weekday` are not `None`. For subclasses of the class `FY5253Mixin` we initialize these parameters in constructor and set them 0 and 1 accordingly.
https://github.com/pandas-dev/pandas/blob/1381f703d834c5bc439c6b5bd3729bc3d6e91c2d/pandas/_libs/tslibs/offsets.pyx#L3579-L3580

Reply:
right, thanks, they can't actually be `None`, so the `is_anchored` definition was too complex to begin with

Reply:
if we check:
```
>>> pd.offsets.FY5253().is_anchored()
FutureWarning: FY5253.is_anchored is deprecated and will be removed in a future version, please use 'obj.n == 1' instead.
True
>>> pd.offsets.FY5253(n=1, startingMonth=1, weekday=1).is_anchored()
True
```
On the other hand for the class Week we have
```
>>> pd.offsets.Week(1).is_anchored()
FutureWarning: Week.is_anchored is deprecated and will be removed in a future version, please use 'obj.n == 1 and obj.weekday is not None' instead.
False
>>> pd.offsets.Week(n=1, weekday=0).is_anchored()
True
```

Reply:
but also, trying to pass `None` just raises
```python
In [12]: FY5253(n=1, weekday=1, startingMonth=1).is_anchored()
<ipython-input-12-37644e23496a>:1: FutureWarning: FY5253.is_anchored is deprecated and will be removed in a future version, please use 'obj.n == 1' instead.
  FY5253(n=1, weekday=1, startingMonth=1).is_anchored()
Out[12]: True

In [13]: FY5253(n=1, weekday=None, startingMonth=None).is_anchored()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[13], line 1
----> 1 FY5253(n=1, weekday=None, startingMonth=None).is_anchored()

File offsets.pyx:3627, in pandas._libs.tslibs.offsets.FY5253Mixin.__init__()

TypeError: an integer is required

In [14]: FY5253(n=1, weekday=None, startingMonth=1).is_anchored()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[14], line 1
----> 1 FY5253(n=1, weekday=None, startingMonth=1).is_anchored()

File offsets.pyx:3628, in pandas._libs.tslibs.offsets.FY5253Mixin.__init__()

TypeError: an integer is required

In [15]: FY5253(n=1, weekday=1, startingMonth=None).is_anchored()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[15], line 1
----> 1 FY5253(n=1, weekday=1, startingMonth=None).is_anchored()

File offsets.pyx:3627, in pandas._libs.tslibs.offsets.FY5253Mixin.__init__()

TypeError: an integer is required
```